disnake/gateway.py

```diff
@@ -638,7 +648,20 @@ async def received_message(self, raw_msg: Union[str, bytes], /) -> None:
         except KeyError:
             _log.debug("Unknown event %s.", event)
         else:
-            func(data)
+            try:
+                func(data)
+            except Exception as e:
+                if self._dispatch_gateway_error is None:
+                    # error handler disabled, raise immediately
+                    raise
+
+                if event in {"READY", "RESUMED"}:  # exceptions in these events are fatal
```

Comment:
I think that hello and reconnect are also fatal, not sure.

Reply:
Those are separate opcodes, this change just takes care of errors in `DISPATCH`ed events.

Reply:
Ah, gotcha